compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -387,13 +398,31 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
 
     @Override
     public void visit(BLangTupleTypeNode tupleTypeNode) {
+        BTypeSymbol typeSymbol = tupleTypeNode.getBType().tsymbol;
+        BSymbol owner = typeSymbol.owner;
+        desugarFieldAnnotations(owner, typeSymbol, tupleTypeNode.members, tupleTypeNode.pos);
         List<BLangSimpleVariable> rewrittenMembers = new ArrayList<>();
         tupleTypeNode.members.forEach(member -> rewrittenMembers.add(rewrite(member, env)));
         tupleTypeNode.members = rewrittenMembers;
         tupleTypeNode.restParamType = rewrite(tupleTypeNode.restParamType, env);
         result = tupleTypeNode;
     }
 
+    private void desugarFieldAnnotations(BSymbol owner, BTypeSymbol typeSymbol, List<BLangSimpleVariable> fields,
```

Comment:
With this approach, and having this be initiated by the ClosureGenerator itself (record and tuple type node visits), the ClosureGenerator is now aware of annotations and how they are modeled. This is not ideal IMO.

Ideally, ClosureGenerator should just be a util class, and the annotation desugar (or even the normal desugar) should handle calling this function appropriately. That'll result in a clean separation of concerns.

Please create an issue, we need to fix this. Specially since the ClosureGenerator is used for multiple things now.

Reply:
Create a separate issue for this(https://github.com/ballerina-platform/ballerina-lang/issues/39345)